cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractAssemblyOperator.java

```diff
@@ -59,7 +60,7 @@
 
     protected final Vertx vertx;
     protected final boolean isOpenShift;
-    protected final AssemblyType assemblyType;
+    protected final ResourceType assemblyType;
```

Comment:
Within the CO `AssemblyType` doesn't really serve a useful function, We could replace with with `String kind` without losing anything.

Reply:
I basically agree. But I think we should keep that for another refactoring PR.